pkg/downloader/chart_downloader.go

```diff
@@ -97,6 +98,13 @@ func atomicWriteFile(filename string, body io.Reader, mode os.FileMode) error {
 	return os.Rename(tempName, filename)
 }
 
+type InMemoryChart struct {
+	Chart                  *bytes.Buffer
+	ProvenanceVerification *provenance.Verification
+}
+
+var InMemoryCache = make(map[string]InMemoryChart)
```

Comment:
Can we make this private and encapsulate the details? Users should not know the details of how the cache is implemented.

Reply:
Done.